File: lista java/at8.java
import java.util.ArrayList;
import java.util.List;


class Empregado {
    private String nome;
    private String cargo;
    private double salario;

    
    public Empregado(String nome, String cargo, double salario) {
        this.nome = nome;
        this.cargo = cargo;
        this.salario = salario;
    }

    
    public void exibirDetalhes() {
        System.out.println("Nome: " + nome + ", Cargo: " + cargo + ", Salário: R$ " + salario);
    }

    public String getNome() {
        return nome;
    }
}


class Empresa {
    private String nome;
    private List<Empregado> empregados;  

    
    public Empresa(String nome) {
        this.nome = nome;
        this.empregados = new ArrayList<>();
    }

    
    public void adicionarEmpregado(Empregado empregado) {
        empregados.add(empregado);
    }

    
    public void exibirEmpregados() {
        System.out.println("Empresa: " + nome + " possui os seguintes empregados:");
        for (Empregado empregado : empregados) {
            empregado.exibirDetalhes();
        }
    }
    
    public static void main(String[] args) {
        
        Empregado empregado1 = new Empregado("Carlos", "Engenheiro de Software", 8000.00);
        Empregado empregado2 = new Empregado("Maria", "Gerente de Projetos", 12000.00);

        
        Empresa empresa = new Empresa("Tech Solutions");

        
        empresa.adicionarEmpregado(empregado1);
        empresa.adicionarEmpregado(empregado2);

       
        empresa.exibirEmpregados();
    }
}

File: lista Go/at8.go
package main

import "fmt"


type Empregado struct {
    Nome   string
    Cargo  string
    Salario float64
}


func (e Empregado) ExibirDetalhes() {
    fmt.Printf("Nome: %s, Cargo: %s, Salário: R$ %.2f\n", e.Nome, e.Cargo, e.Salario)
}


type Empresa struct {
    Nome      string
    Empregados []Empregado  
}


func (e *Empresa) AdicionarEmpregado(empregado Empregado) {
    e.Empregados = append(e.Empregados, empregado)
}


func (e Empresa) ExibirEmpregados() {
    fmt.Printf("Empresa: %s possui os seguintes empregados:\n", e.Nome)
    for _, empregado := range e.Empregados {
        empregado.ExibirDetalhes()
    }
}

func main() {
    empregado1 := Empregado{Nome: "Carlos", Cargo: "Engenheiro de Software", Salario: 8000.00}
    empregado2 := Empregado{Nome: "Maria", Cargo: "Gerente de Projetos", Salario: 12000.00}


    empresa := Empresa{Nome: "Tech Solutions"}

  
    empresa.AdicionarEmpregado(empregado1)
    empresa.AdicionarEmpregado(empregado2)


    empresa.ExibirEmpregados()
}
